src/axom/inlet/ConduitReader.cpp

```diff
@@ -157,6 +157,31 @@ void arrayToMap(const conduit::DataArray<ConduitType>& array,
   }
 }
 
+/*!
+ *******************************************************************************
+ * \brief Recursive name retrieval function
+ * 
+ * \param [in] node The Conduit node to "visit"
+ * \param [out] names The set of paths to add to
+ *******************************************************************************
+ */
+void nameRetrievalHelper(const conduit::Node& node,
+                         std::unordered_set<std::string>& names)
+{
+  // Conduit paths use [0] for array indices, Inlet does not, so they need
+  // to be removed - e.g., foo/[0]/bar vs foo/0/bar
```

Comment:
Is this the only time "[" and "]" can appear in the names? What if I make my json key "foo[bar]"?

Reply:
I've adjusted the container registration logic to check for these characters